### DISCUSSION TITLE
Welcome to Arduino_RT-Thread Discussions!

### DISCUSSION THREAD

**Discussion by onelife** [onelife/Arduino_RT-Thread#6] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by danilliu-sudo** [comment]
Hi, onelife，thank you for you good work!
could you pls provide an interrupt function sample for rt-thread? my hardware is Arduino DUE board. I found there's no interrupt in all the samples.

**Reply by onelife** [reply]
Hi @danilliu-sudo,

Thanks for your interest in the Arduino RT-Thread library project!

The library is trying to utilize as much of the Arduino framework and libraries as possible (to reduce maintenance costs), and most of the common interrupts are handled by the framework. Therefore only a few interrupts are directly handled by the library.

You may find the examples in the driver code. E.g.,
https://github.com/onelife/Arduino_RT-Thread/blob/a56779a313d572b0ebc36fba3be5b4ff957b1be1/src/bsp/LonganNano/drv_usart.c#L208-L242
